File: 06/Java/TripleExpression/Square.java
package TripleExpression;

public class Square implements TripleExpression {
	private TripleExpression value;
    public Square(TripleExpression val) {
        value = val;
    }

    public int evaluate(int x, int y, int z) {
        int a = value.evaluate(x, y, z);
        return a * a;
    }
}

File: 08/CSharp/TripleExpressionApp/TripleExpressionApp/Square.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Square<T> : ITripleExpression<T>
    {
        private ITripleExpression<T> value;
        private IOperator<T> op;
        public Square(ITripleExpression<T> value, IOperator<T> op)
        {
            this.value = value;
            this.op = op;
        }

        public T Evaluate(T x, T y, T z)
        {
            T a = value.Evaluate(x, y, z);
            return op.Square(a);
        }
    }
}
